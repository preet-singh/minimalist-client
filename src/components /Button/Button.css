
/* .NavButton {
    background-color: #cacaca;
    border-radius: 10px;
    border: 0;
    color: black;
    display: block;
    font-size: 12px;
    font-weight: bold;
    height: 45px;
    padding: 8px;
    text-align: center;
    text-decoration: none;
    width: 50px;
  } */

  .NavButton {
    box-shadow:inset 0px 0px 15px 3px #23395e;
    background:linear-gradient(to bottom, #2e466e 5%, #415989 100%);
    background-color:#2e466e;
    border-radius:17px;
    border:1px solid #1f2f47;
    cursor:pointer;
    color:#ffffff;
    font-family:Arial;
    font-size:15px;
    padding:6px 13px;
    text-decoration:none;
    text-shadow:0px 1px 0px #263666;
    display: inline-block;
  }
  
  a.NavButton {
    padding: 16px 8px 8px;
  }
  
  .NavButton:hover {
    cursor: pointer;
  }

  @media screen and (min-width: 320px) {
    .NavButton{
      font-size: 10px;
    }
  }

  @media screen and (min-width: 375px) {
    .NavButton{
      font-size: 11px;
    }
  }

  @media screen and (min-width: 425px) {
    .NavButton{
      font-size: 12px;
    }
  }

  @media screen and (min-width: 768px) {
    .NavButton{
      font-size: 13px;
    }
  }
  