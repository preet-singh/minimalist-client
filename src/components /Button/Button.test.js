import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './Button'

describe(`Button component`, () => {
  const props = {
    tag: 'a',
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a button.Button by default', () => {
    const wrapper = shallow(<Button />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the button from props', () => {
    const wrapper = shallow(<Button {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
