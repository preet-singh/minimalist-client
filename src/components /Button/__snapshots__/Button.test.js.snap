// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a button.Button by default 1`] = `
<a
  className="NavButton "
/>
`;

exports[`Button component renders the button from props 1`] = `
<a
  className="NavButton test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</a>
`;
